src/python/WMCore/MicroService/Unified/MSTransferor.py

```diff
@@ -176,16 +178,63 @@ def getRequestRecords(self, reqStatus):
 
         return orderedRequests
 
+    def checkDataPlacement(self, wflow):
+        """
+        Check which data is already in place (in the workflow SiteWhitelist)
+        :param wflow: workflow object
+        :return: don't know yet
```

Comment:
Since I don't know what you expect this method to do, but I think you should decomposed it into pieces. One method should check a single dataset placement. Then this method should get list of datasets and ask previous method to check their placements. The return would be either list of dict of datasets along with their checks.

Reply:
I updated the docstring to make it more clear. This method is pretty straightforward though and all the information is already available in the `Workflow` object. 
It simply does an intersection of the site list with the block/dataset location, if it evaluates to True, block/dataset gets popped out of the structure/dict.